adr/ADR-15.md

```diff
@@ -0,0 +1,233 @@
+# JetStream Subscribe Workflow
+
+|Metadata|Value|
+|--------|-----|
+|Date    |2021-08-11|
+|Author  |@kozlovic|
+|Status  |Partially Implemented|
+|Tags    |jetstream, client|
+
+## Context
+
+This document attempts to describe the workflow of a JetStream subscription in the library, from the creation, the behavior at runtime, to the deletion of a subscription.
+
+## Design
+
+### Creation
+
+The library should have API(s) that allow(s) creation of a JetStream subscription. Depending on the language, there may be various type of subscriptions (like Channel based in Go), or with various options such as Queue subscriptions, etc... There can be different APIs or one with a set of options.
+
+#### Configuration
+
+When creating a subscription, the user can provide several things:
+
+- A subject that is in some cases required (see "Stream name" section below)
+- An optional stream name
+- An optional consumer name
+- An optional JS consumer configuration (see below), if the library is expected to create the JS consumer
+- A Queue name if this subscription is meant to be a queue subscription
+- Indication if this is a "Pull" subscription or not
+- An optional way of indicating the the given subject will be used to create the internal subscription, by-passing any lookup/consumer creation. For the purpose of this document, let's call this option `SubjectIsDelivery`.
```

Comment:
This is confusing. By the last sentence is this something other than 'bind'?
Also where is the `Indication if this is a "Bind"` part. Bind was a previously agreed upon term and has already been coded (although not released)

Reply:
Bind is when you specify a stream and consumer name. It has been done in Go already in the already released v1.11.0.

Reply:
Yes, this option is to say that the library will go directly create the NATS subscription on the subject passed to the "subscribe" API call, and won't do any creation of consumer/lookup/or any check.

Reply:
This new option is removed.